packages/ui-react/src/components/UserMenu/UserMenu.tsx

```diff
@@ -1,104 +1,89 @@
+      </div>
+    );
+  }
 
   return (
-    <>
-      <h2 className={styles.sectionHeader} id={titleId}>
-        {t('nav.settings')}
-      </h2>
-      <ul className={styles.menuItems}>
-        {currentProfile && (
-          <li>
-            <MenuButton
-              small={small}
-              onClick={onButtonClick}
-              to={userProfileURL(currentProfile.id ?? '')}
-              label={t('nav.profile')}
-              startIcon={<ProfileCircle src={currentProfile?.avatar_url} alt={currentProfile?.name ?? ''} />}
-            />
-          </li>
-        )}
-        <li>
-          <MenuButton
-            small={small}
-            onClick={onButtonClick}
-            to={PATH_USER_ACCOUNT}
-            label={t('nav.account')}
-            startIcon={<Icon icon={AccountCircle} />}
-            tabIndex={tabIndex}
-          />
-        </li>
-        {favoritesEnabled && (
-          <li>
-            <MenuButton
-              small={small}
-              onClick={onButtonClick}
-              to={PATH_USER_FAVORITES}
-              label={t('nav.favorites')}
-              startIcon={<Icon icon={Favorite} />}
-              tabIndex={tabIndex}
-            />
-          </li>
-        )}
-        {showPaymentsItem && (
-          <li>
-            <MenuButton
-              small={small}
-              onClick={onButtonClick}
-              to={PATH_USER_PAYMENTS}
-              label={t('nav.payments')}
-              startIcon={<Icon icon={BalanceWallet} />}
-              tabIndex={tabIndex}
-            />
-          </li>
-        )}
-        <li className={classNames(styles.divider, { [styles.small]: small })}>
-          <MenuButton small={small} onClick={onLogout} label={t('nav.logout')} startIcon={<Icon icon={Exit} />} tabIndex={tabIndex} />
-        </li>
-      </ul>
-    </>
+    <div>
+      <HeaderActionButton
+        aria-label={t('open_user_menu')}
+        aria-controls="menu_panel"
+        aria-expanded={open}
+        aria-haspopup="menu"
+        onClick={onOpen}
+        onBlur={onClose}
+      >
+        {profilesEnabled && profile ? <ProfileCircle src={profile.avatar_url} alt={profile.name || t('profile_icon')} /> : <Icon icon={AccountCircle} />}
+      </HeaderActionButton>
+      <Popover className={styles.popover} isOpen={open} onClose={onClose}>
+        <Panel id="menu_panel">
+          <div onFocus={onOpen} onBlur={onClose}>
+            {profilesEnabled && (
+              <ProfilesMenu
+                onButtonClick={onClose}
+                profiles={profiles ?? []}
+                currentProfile={profile}
+                selectingProfile={profileLoading}
+                selectProfile={onSelectProfile}
+                small
+              />
+            )}
+            <UserMenuNav focusable={open} onButtonClick={onClose} showPaymentItems={true} currentProfile={profile} favoritesEnabled={favoritesEnabled} small />
```

Comment:
Should we use a dynamic value for `showPaymentItems`?

Reply:
It was already there 😅